device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/amqps/AmqpsCbsSenderLinkHandler.java

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package com.microsoft.azure.sdk.iot.device.transport.amqps;
+
+import com.microsoft.azure.sdk.iot.device.DeviceClientConfig;
+import com.microsoft.azure.sdk.iot.device.Message;
+import com.microsoft.azure.sdk.iot.device.exceptions.TransportException;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.qpid.proton.Proton;
+import org.apache.qpid.proton.amqp.messaging.AmqpValue;
+import org.apache.qpid.proton.amqp.messaging.ApplicationProperties;
+import org.apache.qpid.proton.amqp.messaging.Properties;
+import org.apache.qpid.proton.amqp.messaging.Section;
+import org.apache.qpid.proton.engine.Sender;
+import org.apache.qpid.proton.message.impl.MessageImpl;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * Every SAS token based authentication over AMQP requires a CBS session with a sender and receiver link. This
+ * class defines the sender link which proactively sends renewed sas tokens to keep the device sessions authenticated.
+ */
+@Slf4j
+public final class AmqpsCbsSenderLinkHandler extends AmqpsSenderLinkHandler
+{
+    private static final String SENDER_LINK_ENDPOINT_PATH = "$cbs";
+
+    private static final String SENDER_LINK_TAG_PREFIX = "cbs-sender";
+
+    private static final String CBS_TO = "$cbs";
+    private static final String CBS_REPLY = "cbs";
+
+    private static final String OPERATION_KEY = "operation";
+    private static final String TYPE_KEY = "type";
+    private static final String NAME_KEY = "name";
+
+    private static final String OPERATION_VALUE = "put-token";
+    private static final String TYPE_VALUE = "servicebus.windows.net:sastoken";
+
+    private static final String DEVICES_PATH = "/devices/";
+
+    AmqpsCbsSenderLinkHandler(Sender sender, AmqpsLinkStateCallback amqpsLinkStateCallback)
+    {
+        super(sender, amqpsLinkStateCallback, UUID.randomUUID().toString());
+
+        this.senderLinkTag = SENDER_LINK_TAG_PREFIX;
+        this.senderLinkAddress = SENDER_LINK_ENDPOINT_PATH;
+    }
+
+    static String getCbsTag()
+    {
+        return SENDER_LINK_TAG_PREFIX;
+    }
+
+    @Override
+    public String getLinkInstanceType()
+    {
+        return "cbs";
```

Comment:
you can use CBS_REPLY  here

Reply:
I'll add a private static final variable called LINK_TYPE on all these sender/receiver links instead, so they all feel similar